src/main/java/net/citizensnpcs/npc/CitizensNPC.java

```diff
@@ -185,10 +180,9 @@ public boolean spawn(Location at) {
         at = at.clone();
         getTrait(CurrentLocation.class).setLocation(at);
 
-        entityController.spawn(at, this);
+        boolean couldSpawn = entityController.spawn(at, this);
 
         net.minecraft.server.v1_8_R3.Entity mcEntity = ((CraftEntity) getEntity()).getHandle();
-        boolean couldSpawn = !Util.isLoaded(at) ? false : mcEntity.world.addEntity(mcEntity, SpawnReason.CUSTOM);
```

Comment:
I was trying to send the playerlist packet after the entity was added to the world but couldn't do that easily from the controller spawn method since it happened. TBH, I'm not 100% sure if it even matters. I can try putting it back and doing more tests.